src/test/scala/linting/rule/LintConflictingModuleNamesSpec.scala

```diff
@@ -0,0 +1,136 @@
+
+import firrtl._
+import firrtl.ir.Port
+import firrtl.annotations._
+import firrtl.testutils.{FirrtlMatchers, FirrtlPropSpec}
+
+case class StableNameAnnotation(target: IsModule) extends SingleTargetAnnotation[IsModule] {
+  def duplicate(newTarget: IsModule): StableNameAnnotation = this.copy(target = newTarget)
+}
+
+case class UnstableNameAnnotation(target: IsModule) extends SingleTargetAnnotation[IsModule] {
+  def duplicate(newTarget: IsModule): UnstableNameAnnotation = this.copy(target = newTarget)
+}
+
+class RenameDesiredNamesSpec extends FirrtlPropSpec with FirrtlMatchers {
+  val transform = new RenameDesiredNames
+
+  case class TestCase(
+    input: String,
+    annos: Seq[Annotation]
+  )
+
+  def stabilizeNames(testCase: TestCase): CircuitState = {
+    stabilizeNames(CircuitState(Parser.parse(testCase.input), UnknownForm, testCase.annos))
+  }
+
+  def stabilizeNames(state: CircuitState): CircuitState = {
+    transform.runTransform(state)
+  }
+
+  def test(testCases: TestCase *): Unit = {
+    val firstState = stabilizeNames(testCases.head)
+    val firstUnstableNames = firstState.annotations.collect {
+      case a: UnstableNameAnnotation => a
+    }
+    val firstStableNames = firstState.annotations.collect {
+      case a: StableNameAnnotation => a
+    }
+
+    testCases.tail.foldLeft(firstStableNames, firstUnstableNames) {
+      case ((stable, unstable), testCase) =>
+        val state = stabilizeNames(testCase)
+        val currUnstableNames = state.annotations.collect {
+          case a: UnstableNameAnnotation => a
+        }
+        val currStableNames = state.annotations.collect {
+          case a: StableNameAnnotation => a
+        }
+
+        currStableNames should be (stable)
+        currUnstableNames should not be (unstable)
+        (currStableNames, currUnstableNames)
+    }
+  }
+
+  property("It should rename modules to desired names if only one module") {
+    val top = CircuitTarget("Foo")
+    test(
+      TestCase(
+        """|circuit Foo:
+           |  module Bar_1:
+           |    output in1: UInt<1>
+           |  module Bar_2:
+           |    output in1: UInt<1>
+           |    output in2: UInt<1>
+           |  module Bar_3:
+           |    output in1: UInt<1>
+           |    output in2: UInt<1>
+           |    output in3: UInt<1>
+           |  module Foo:
+           |    inst bar_1 of Bar_1
+           |    inst bar_2 of Bar_2
+           |    inst bar_3 of Bar_3
+           |""".stripMargin,
+        Seq(
+          UnstableNameAnnotation(top.module("Bar_1")),
+          StableNameAnnotation(top.module("Bar_2")),
+          StableNameAnnotation(top.module("Bar_3")),
+          OverrideDesiredNameAnnotation("Bar_2", top.module("Bar_2")),
+          OverrideDesiredNameAnnotation("Bar_3", top.module("Bar_3"))
+        )
+      ),
+      TestCase(
+        """|circuit Foo:
+           |  module Bar_1:
+           |    output in1: UInt<2>
+           |  module Bar_5:
```

Comment:
I don't get why this is what this test does. What is the test showing / verifying? Didn't we say we wanted it called Bar_2?

Can we use/add more real-world examples, like "BarWith2Inputs"?

Reply:
It is supposed to show that between different runs of firrtl, the names stay the same or change. so the test is testing that `Bar_2`, `Bar_3` exist in the first and second runs because of the `OverrideDesiredName` annotations. yeah I'll add more tests, these types of tests aren't as useful with only the exact naming strategy

Reply:
But. where are Bar_5 and Bar_6 coming from...?

Reply:
never mind i realize that these are coming from the annotations below